Fix sunz normalization for viirs_compact

ninahakansson opened the pull request:
SDR data has sun_zenith_angle_correction_applied as default, viirs_compact does not! We now read data with the sunz_corrected modifier and satpy applyes the normalization. Note that reflective bands are
 read separately to avoid sunz_correction applied to the IR channels.

<!-- Describe what your PR does, and why -->

 - [x] Closes #107
 - [ ] Tests added <!-- for all bug fixes or enhancements -->
 - [x] Tests passed: Passes ``pytest level1c4pps`` <!-- for all non-documentation changes) -->
 - [x] Passes ``flake8`` <!-- remove if you did not edit any Python files -->

ninahakansson committed:
Fix sunz normalization for viirs_compact

SDR data has sun_zenith_angle_correction_applied as default,
viirs_compact does not! We now read data with the sunz_corrected modifier
and satpy applyes the normalization. Note that reflective bands are
 read separately to avoid sunz_correction applied to the IR channels.

BengtRydberg commented on level1c4pps/viirs2pps_lib.py:
```diff
@@ -114,8 +114,11 @@ def set_header_and_band_attrs(scene, orbit_n=0):
+        # SDR has data has sun_zenith_angle_correction_applied, viirs_compact does not!
+        # We now read data with the sunz_corrected modifier
         scene[band].attrs['sun_zenith_angle_correction_applied'] = 'True'
+        if "sunz_corrected" not in scene['M05'].attrs['modifiers']:
```
Why checking just M05? Maybe more clear if you move this check to line 156 and do an assert there (if we do not trust satpy):

```assert "sunz_corrected" in scene['M05'].attrs['modifiers']```

Or I would recommend that you remove this check and add a new input parameter named `sun_zenith_angle_correction_applied` to this function so it looks something like this: 

```
def set_header_and_band_attrs(scene, sun_zenith_angle_correction_applied, orbit_n=0):
    if sun_zenith_angle_correction_applied:
        scene[band].attrs['sun_zenith_angle_correction_applied'] = 'True'
```
Then you can probably remove the comments on line 117 and 118 as well. To me comments is an indication that the code can be written more clear than presently. 

and e.g. one line 156 and 159 you can add the following line:

`sun_zenith_angle_correction_applied = True`

ninahakansson committed:
Fix test

Use the modifiers to set the sunz-correction attribute already in the main attribute function.

ninahakansson committed:
Fix syntax
